server/reportConfiguration/constants/types.ts

```diff
@@ -83,3 +83,8 @@ export type Type = (typeof types)[number]['code']
  * @deprecated
  */
 export type NomisType = (typeof types)[number]['nomisCode']
+
+/** Look up an incident type by code */
+export function getTypeDetails(code: string): (typeof types)[number] | null {
```

Comment:
Perhaps we can have a `TypeDetails` type set as

```Typescript
export type TypeDetails = (typeof types)[number]
``` 

`Type` becomes:

```Typescript
export type Type = TypeDetails['code']
```

and this function signature becomes:

```Typescript
export function getTypeDetails(code: string): TypeDetails | null {
}
```

overkill? Besides it's a bit odd that the return type of this function here is the specific union of all the valid report types rather than a simpler

```Typescript
export type TypeDetails = {
  code: string
  description: string
  active: boolean
  nomisCode: string
}
```

but it's probably more reflective of the valid report types allowed (and it would evolve over time as more types are added or deactivated)

Reply:
(just an observation that `(typeof types)[number]` is used twice and `TypeDetails` may be a nice name for it but you don't need to necessarily change anything)